fondant/component.py

```diff
@@ -182,9 +184,9 @@ def _load_or_create_manifest(self) -> Manifest:
     def transform(self, dataframe: dd.DataFrame, **kwargs) -> dd.DataFrame:
         """Abstract method for applying data transformations to the input dataframe."""
 
-    def _process_dataset(self, dataset: FondantDataset) -> dd.DataFrame:
+    def _process_dataset(self, dataset: DaskDataLoader) -> dd.DataFrame:  # type: ignore[override]
```

Comment:
Seems like `dataset` is defined as an optional parameters in the parent class but we're enforcing is as a mandatory in the child class. Maybe it's better to change 
```
 def _process_dataset(self, dataset: FondantDataset) 
 ```
 
 to 
 
 ```
 def _process_dataset(self, **kwargs) 
 ```
 In the parent class.

Reply:
I think we can get around this issue if we implement the `run()` method as follows:

```
class Component:

    def _write_dataset(dataframe, *, manifest):
        output_dataset = DaskDataWriter(output_manifest)

        # write index and output subsets to remote storage
        output_dataset.write_index(dataframe)
        output_dataset.write_subsets(dataframe, self.spec)

    def run(self):
        """Runs the component."""
        input_manifest = self._load_or_create_manifest()
  
        output_df = self._process_data(input_manifest)
  
        output_manifest = input_manifest.evolve(component_spec=self.spec)
  
        self._write_dataset(output_df, manifest=output_manifest)
  
        self.upload_manifest(output_manifest, save_path=self.args.output_manifest_path)

class LoadComponent:

    def _process_data(manifest):
        return self.load(...)

class TransformComponent:

    def _process_data(manifest):
         dataset = DaskDataLoader(manifest)
         df = dataset.load_dataframe(self.spec)
         df = self.transform(df, **self.user_arguments)
```

Reply:
indeed this seems more appropriate :) thanks for the suggestion

Reply:
It's a bit weird to me to call the argument `dataset` and then specify a `DataLoader` class. Would then also call the argument `dataloader`.